### PR TITLE
Support ABTest cluster provider and a ABTest case

### PR DESCRIPTION
**Pull request by you06**
This PR enable abtest and add k8s provider for double cluster.

### CODE REVIEW

**Comment by you06** [comment]
In my test, it leads error very quickly due to the data inconsistency. The log looks weird and I still need some time to find out why it failed.

File: pkg/test-infra/abtest/operation.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+// Apply abtest cluster
+func (t *Ops) Apply(tc *Recommendation) error {
+	var (
+		wg   sync.WaitGroup
+		errs []error
+	)
+	wg.Add(2)
+
+	go func() {
```
Comment (mahjonp) [inline pkg/test-infra/abtest/operation.go]: use [errgroup.Group{}](https://godoc.org/golang.org/x/sync/errgroup) here?

File: pkg/test-infra/fixture/fixture.go [hunk 2]
```diff
@@ -165,6 +166,9 @@ func init() {
 	flag.StringVar(&Context.LokiAddress, "loki-addr", "", "loki address. If empty then don't query logs from loki.")
 	flag.StringVar(&Context.LokiUsername, "loki-username", "", "loki username. Needed when basic auth is configured in loki")
 	flag.StringVar(&Context.LokiPassword, "loki-password", "", "loki password. Needed when basic auth is configured in loki")
+
+	flag.StringVar(&Context.ABTestConfig.Cluster1Version, "cluster1-version", "", "specify version for cluster1")
```
Comment (mahjonp) [inline pkg/test-infra/fixture/fixture.go]:
```suggestion
	flag.StringVar(&Context.ABTestConfig.Cluster1Version, "abtest.a.image-name", "", "specify version for cluster1")
```

**Comment by mahjonp** [comment]
Rest LGTM

File: pkg/test-infra/fixture/fixture.go (hunk 2 above)
Comment (you06) [reply pkg/test-infra/fixture/fixture.go]: I think "abtest.a.version" would be better, cause it only defines the cluster version but not the whole image name.

Comment (mahjonp) [reply pkg/test-infra/fixture/fixture.go]: OK for me

**Comment by you06** [comment]
@mahjonp @mapleFU PTAL again

**Pull request merged**